### ISSUE TITLE
about cos_theta

### ISSUE THREAD

**Issue by alexliyang** [Joker316701882/Additive-Margin-Softmax#2]
Hi :
    I have question :  in order to get the cos_theta , here  tf.matmul is used, when I train a net, "nan" log is showed, so I confused, tf.matmul get a matrix result, tf.multiply get a scalar value. 
    Can I use tf.multiply here ?

**Comment by Joker316701882** [comment]
@alexliyang 
Use tf.multiply will need one or two more lines to implement this. But there is no need and no reason to use tf.multiply. There must be other problems cause u get "nan" value but not tf.matmul : )

**Comment by alexliyang** [comment]
thank you very much!
I reference your code to write arcface (insightface) loss function. I correct some errors and start trainning .
I hope 'nan' log disappered.

**Comment by Joker316701882** [comment]
@alexliyang 
Oh great! I also tried Arcface once, and failed to converge. Looking forward to your progress on that : )

**Comment by alexliyang** [comment]
in fact, we try to use AM (mobilenet, tf), but the converge is very slow and acc is not good, only about 0.74 .
we can not understand where the error is .

**Comment by Joker316701882** [comment]
@alexliyang 
See [Issue](https://github.com/Joker316701882/Additive-Margin-Softmax/issues/1), there are some useful tips in it, hope it helps u.
